server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -293,12 +296,62 @@ public int hashCode() {
         }
     }
 
-    private static TopKState topKState(RamAccounting ramAccounting, int limit) {
+    record TopKLongState(LongsSketch sketch, int limit) implements State {
+
+        static long SHALLOW_SIZE = RamUsageEstimator.shallowSizeOfInstance(TopKLongState.class);
+
+        public List<Map<String, Object>> result() {
+            if (sketch.isEmpty()) {
+                return List.of();
+            }
+            LongsSketch.Row[] frequentItems = sketch.getFrequentItems(ErrorType.NO_FALSE_NEGATIVES);
+            int limit = Math.min(frequentItems.length, this.limit);
+            var result = new ArrayList<Map<String, Object>>(limit);
+            for (int i = 0; i < limit; i++) {
+                var item = frequentItems[i];
+                result.add(Map.of("item", item.getItem(), "frequency", item.getEstimate()));
+            }
+            return result;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            TopKLongState topKLongState = (TopKLongState) o;
+            return limit == topKLongState.limit && Objects.equals(result(), topKLongState.result());
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(result(), limit);
+        }
```

Comment:
Could consider making `State` abstract and move `equals`/`hashCode` into it, to avoid having to repeat the boilerplate. I assume with double/float, etc.. we'll have even more variants.